### ISSUE TITLE
multiple "supports" 

### ISSUE THREAD

**Issue by AkandaAshraf** [tkipf/gcn#54]
Hi 

I was wondering about the "num_supports=1". 

Is it possible to input multiple supports? Let's say for a graph with different types of connections can I input two different adj matrices / supports?

           support = list()
           support.append(sp.coo_matrix(adj1))
           support.append(sp.coo_matrix(adj2))

It looks like that the "sparse_to_tuple" function is written to deal with list inputs also the convolve operation seem to have (layers.py line#240):

            supports.append(support)
        output = tf.add_n(supports)

Did you intend to have this feature which is to be able to deal with two different adj matrices for a same graph?

I tried to pass a support with a list of two adj matrices, however I am getting an error when I tried to do this:  

  File "D:/gcn/gcn-master/gcn/akanda.py", line 488, in experiment
    outs = sess.run([model.opt_op, model.loss, model.accuracy], feed_dict=feed_dict)
  File "C:\Program Files\Python36\lib\site-packages\tensorflow\python\client\session.py", line 900, in run
    run_metadata_ptr)
  File "C:\Program Files\Python36\lib\site-packages\tensorflow\python\client\session.py", line 1104, in _run
    np_val = np.asarray(subfeed_val, dtype=subfeed_dtype)
  File "C:\Program Files\Python36\lib\site-packages\numpy\core\numeric.py", line 492, in asarray
    return array(a, dtype, copy=False, order=order)
ValueError: setting an array element with a sequence. 

Thanks a lot!

**Comment by tkipf** [comment]
Yes this feature was intended. Unfortunately the API of tensorflow changed
in the meantime which gives you this error I suppose. You might want to try
feeding the adjacency matrices as we did in the ‚recommender-gcn‘
repository by Rianne van den Berg (I currently don’t have Internet access
while writing this, but if you look this up on Google you should find it).
Hope this helps!

**Comment by guotong1988** [comment]
multiple "support" means multiple graph?

**Comment by tkipf** [comment]
Each ‘support’ is an adjacency matrix, so you can see it as multiple
graphs, yes.

**Comment by guotong1988** [comment]
Thx for your reply!

**Comment by hoeikhwang** [comment]
I'm trying to understand the naming sense behind 'support'.
Is there some relation with 'support of random variable' in probability theory?
